src/components/container/AIAssistant/AIAssistant.tsx

```diff
@@ -86,10 +87,34 @@ export default function (props: AIAssistantProps) {
                     let toolInput = streamEvent.data.input.input;
 
                     if (props.debug) {
-                        formatCode(toolName, toolInput)
-                            .then((formattedMessage) => {
-                                addToolMessage(streamEvent.run_id, "```js\n" + formattedMessage + "```");
-                            });
+                        let formattedMessage = await formatCode(toolName, toolInput)
+                        addMessage(streamEvent.run_id, "```js\n" + formattedMessage + "```", "tool");
+                    }
+
+                    if (toolName === "graphing") {
+                        addMessage(streamEvent.run_id, `data:image/png;base64,${streamEvent.data.output.content}`, "image");
```

Comment:
This shouldn't take for granted that the graphing tool has produced a viable image. If it fails right now, this is still run but streamEvent.data.output.content is undefined.
![image](https://github.com/user-attachments/assets/45169190-2a54-427a-9f1e-fb5b5b929664)